pkg/tfgen/generate.go

```diff
@@ -1874,13 +1873,20 @@ func getNestedDescriptionFromParsedDocs(entityDocs entityDocs, path docsPath) (s
 		//
 		// (There may be other, unknown, reasons why this behavior exists.)
 		//
+		// Additionally, a lot of nested type descriptions are listed under the "Attributes Reference" section and as such
+		// will have been parsed into entityDocs.Attributes. In the AWS provider, this behavior is responsible for many
+		// type property descriptions.
+		//
 		// In case #1 above, we are generating an incorrect schema because the upstream schema is incorrect, and we would
 		// arguably be better off not having any description in our docs. In case #2 above, this is fairly risky fallback
 		// behavior with may result in incorrect docs, per pulumi-terraform-bridge#550.
 		//
 		// We should work to minimize the number of times this fallback behavior is triggered (and possibly eliminate it
 		// altogether) due to the difficulty in determining whether the correct description is actually found.
-		return attribute, true
+		if description, ok := entityDocs.Attributes[string(attrPath)]; ok {
+			return description, true
+		}
```

Comment:
I don't see where this code section is under test. Can you add a test here?

Reply:
Added.